### DISCUSSION TITLE
Virtual camera output

### DISCUSSION THREAD

**Discussion by riemsesy** [sircharlo/meeting-media-manager#270] [Ideas] [answered]
Hello brother Charles,

Nice piece of software. Thanks for taking the time to write this and sharing it with us.

We use OBS https://github.com/obsproject/obs-studio to play all media.
Of course, I have to download all JW media manually. 
Originally I have a script to rename all files to mp4 so that vlc will play the pictures too, because I copy each file to a numbered folder.
From within OBS I can call VLC and play the folders per number. So a bit of manual labor, but quickly done. 
Now I will use JWMMF :-)

The big big advantage why I use OBS is that it has a virtual camera. Our camera is linked to OBS and all media which is played via OBS is  streamed to the virtual OBS camera. I then select the virtual OBS camera in Zoom and our brothers and sisters can see the speaker or the media that is played.

JWMMF can play on a second screen, and I guess br's use Zoom share to stream it to brothers and sisters at home.
Wouldn't it be nice if you can / may use the virtual cam function like OBS has. 
And if you can implement camera capturing too then JWMMF would be a one tool for everything in the congregation.

**Comment by Chipperchap** [comment]
Hi @riemsesy 
INTERESTING POST!  What is the view like for the ZOOM participants at home when you use the virtual camera? Do they find the virtual camera a bit laggy/glitchy? I find that when I use the virtual camera it looks like it is always trying to play catch-up with movements, especially quick movements such as hand gestures.  Have you found similar at all or do the friends just get used to it... don't know anything different?

**Reply by riemsesy** [reply]
Hi @chipperchap, 
We installed a PTZ cam a clone brand from ptzoptics. The cam has NDI SDI and HDMI connections. 
https://www.minrraycam.com/uv570-series-ndi-full-hd-ptz-camera.html we have the 570-20
There are many Chinese look-a-likes: rgblink, smtav
It's connected to the PC with https://www.blackmagicdesign.com/products/decklink/techspecs/W-DLK-06

We noticed some lag with the NDI connection over the UTP cable/network. The other choice to connect was HDMI, but we skipped HDMI because of the distance from the cam to pc. A cable to connect the cam to the PC with HDMI over fibre 35 meters is very expensive. For a short distance, HDMI is the best way to connect. 
So we used a coax SDI cable combined with the DeckLink SDI / HDMI PCIe card to connect the cam to the pc. We experience no lag with this connection and the audio and video are in sync. Our PC was old so we replaced the motherboard and CPU with an intel i5 12400f (no integrated CPU) and Nvidia GTX 1650 graphics card. 

The cam can generate 1080 HD, but I've configured the cam just to stream 720p, that's enough and uses fewer resources.

**Reply by sircharlo** [reply]
This is in fact very interesting. I'm looking into **maybe** adding an OBS-Studio compatibility mode to JWMMF. 

Something along the lines of:
- Configure two scenes in OBS Studio, named for example "Media" (JWMMF media window capture, no cursor) and "Camera" (video device capture)
- Configure `obs-websocket` to allow connections from port #### with password XXXX
- Add configurable settings in JWMMF to enable/disable OBS Studio compatibility mode, along with configurable port number and password for `obs-websocket`

If OBS Studio compatibility mode is enabled, then JWMMF would automatically:
- Activate the "Camera" scene using `obs-websocket` when media presentation mode is started in JWMMF, and when the "stop" button is pressed on any media
- Activate the "Media" scene using `obs-websocket` when the "play" button is pressed on any media

I'll have to do some tests and setup, as our KH doesn't currently use OBS Studio, but it's definitely promising. This method would completely avoid having to screen share in Zoom.

**Reply by joaochris** [reply]
@sircharlo If you can get this it will be amazing! I think this is pretty tricky to do, but it's worth it. I think the most complicated would be in the videos where we don't need to hit the "stop" button. But just the virtual camera output would help.😅😏

**Reply by sircharlo** [reply]
Just tested a quick and dirty implementation of this. Works absolutely great.
Stay tuned.

**Reply by riemsesy** [reply]
@sircharlo yes we don't use zoom share for video. 
We do have audio share enabled because streaming kingdom songs and other music through the mic sounds muffled. With audio share enabled it's crystal clear. 

It really sounds very good how you think this can be set up with a web socket connection.

**Reply by sircharlo** [reply]
If anyone is comfortable with git and node/npm, I've just pushed a commit that should enable preliminary OBS Studio support. Clone the repo, then run `npm i; npm start`. 

The `obs-websocket` plugin is required in OBS Studio for this to work; I'd love some feedback.

![image](https://user-images.githubusercontent.com/1275666/165965508-f0ffce2e-7538-4e0b-8cae-7769e59419b7.png)

**Reply by joaochris** [reply]
@sircharlo I will test it now! I'll be right back to give feedback.

**Reply by joaochris** [reply]
https://user-images.githubusercontent.com/76891864/165978744-f7525433-4dba-489a-af54-ff030c3d70b7.mp4

@sircharlo It works perfectly!! 🎉😎 Congratulations on your intelligence and commitment. I have a suggestion if it's feasible. Would it be possible to put the part where we define the names of the scenes in the media display mode?
This gave us the possibility to instantly change the Camera scene (in case we use the same camera in several scenes) and not leave the media display mode.

**Reply by riemsesy** [reply]
I have no coding experience, i would love to test it :-)

**Reply by joaochris** [reply]
@riemsesy Is not difficult! The steps I did were:
1. First I downloaded the **GIT**: https://git-scm.com/
2. Then the **NodeJS LTS version**: https://nodejs.org/en/download/
3. Installed both
4. Then I opened git bash and put `git clone https://github.com/sircharlo/jw-meeting-media-fetcher`
5. Then I navigated to the folder at C:\Users\[YOUR USER]\jw-meeting-media-fetcher with the right mouse button and then _Git Bash Here_
6. Then `npm i`
Wait for it to install all modules
7. Then `npm start`
After this last command the program starts

If you want to reopen the program, just follow point 5 and 7.
To download project updates with point 5 done, it's `git pull`


Note: Install OBS-Websocket https://github.com/obsproject/obs-websocket/releases/tag/4.9.1
and then configure it in the tools tab of OBS Studio by putting a password.

**Reply by sircharlo** [reply]
Good point @joaochris . New commit; once again, very early testing. Nothing pretty yet, just testing functionality!
Let me know what you think.

**Reply by joaochris** [reply]
@sircharlo Initially, I couldn't get it to work. But I realized that I had to open OBS first and then start JWMMF.
I have a suggestion: In the part of the Media Window field, when it was typed or scene name in the settings did not appear in the media display mode. (Only the other scenes).

**Reply by Chipperchap** [reply]
Okay, followed all the steps above and got it working. (Very difficult for my aged and currently Covid-19 positive brain!!!) 
(BTW the default password for the plugin is "Changeme". Forgot to change it so had to find it!)
 
QN 1: So I take it that the ZOOM host will *always* have share computer audio on right?  Or just share when the songs are playing?

QN 2: The quality of the picture depends on the quality of the computer graphics generator machine (don't know what the name is)? I find mine a bit blurry... possibly because I only have the usual onboard graphics thingee that comes with my PC. Will that limit how useful this feature is...?

 QN: 3 OBS still needs to be running, but can be put in minimize mode so that it is out of the way of the screen real-estate... is that right? ...or will JWMMF 'take over' OBS as a virtual camera?  (Probably should stop asking questions!)

**Reply by joaochris** [reply]
@Chipperchap 

Great!

1 - Yes, it can be :)
2 - Yes, if you have a video card with support for NVENC or VCE the quality will be better! Also, on the zoom.us website logged into the account, you can test the Group HD video option, between 720P or 1080P resolutions.
3 - Yes, OBS needs to be running, but it can be minimized.

**Reply by riemsesy** [reply]
QN 1: indeed. it works without audio sharing on, just a little blunt/muffled
QN 2: and or maybe the camera. it really depends on your setup. quality of cam, processing power of PC. An other congregation in our region, I know they have a Dell Optiplex Micro 3070 also with integrated graphics controller in the CPU. you don't need the latest CPU or dedicated GPU (graphic processing unit)
QN 3: Yes OBS will send the camera image and media to zoom. Combined with JWMMF you probably can minimize OBS. Especially when you have one camera setting for a view of the complete stage. We have different camera settings, for the speaker, the reader, the table and complete stage.

Never stop asking questions... I've had covid twice, and the last time it really affected my cognitive functions.

**Reply by riemsesy** [reply]
we should start a chat group :-D

**Reply by riemsesy** [reply]
@joaochri
s I saw you used windows capture. don't you get a yellow line around the window?

![image](https://user-images.githubusercontent.com/39587825/166072243-621d8300-703d-4f35-9131-0d13f4af6889.png)

**Reply by joaochris** [reply]
@riemsesy So, it doesn't happen to me, but it happened to a brother here. I solved it like this: **right click on the desktop** > **Display Settings** > **Graphics Settings** > _Browse_ (Find the obs studio folder) and open the **obs64.exe** file.
Once selected it will appear in the list. Click on it, then on options and choose **energy saving**. Then open OBS and in the capture method select the one from the print below (in english).

![Screenshot_4](https://user-images.githubusercontent.com/76891864/166073315-c571a126-2e22-4966-a65d-62f6314f4ae9.png)

**Reply by riemsesy** [reply]
I tried it, but it the yellow line stays. I am testing on my own notebook. in the kingdom hall we use gamecapture any full screen. works perfect when media is pushed to the TV's (we have beamers).  I count on it it will work in the KH.

**Reply by joaochris** [reply]
@riemsesy Did you use window or display capture?
When I use window capture it also appears. 

![Screenshot_5](https://user-images.githubusercontent.com/76891864/166077470-fc46ff29-f18c-4033-a775-4968870e31e8.png)

But for the test I used a virtual secondary screen and display capture. If in the display capture the yellow list appears, what I explained above should solve it.

**Reply by sircharlo** [reply]
> I realized that I had to open OBS first and then start JWMMF. 

I'll try to make this less true (find OBS even if it's opened after JWMMF).

> I have a suggestion: In the part of the Media Window field, when it was typed or scene name in the settings did not appear in the media display mode. (Only the other scenes).

Yup. Coming soon.

**Reply by sircharlo** [reply]
> QN 1: So I take it that the ZOOM host will _always_ have share computer audio on right? Or just share when the songs are playing?

In our KH, the audio from the computer is sent to the analog mixer. Then, all audio from the mixer (mics, computer sound, etc) is sent *back* to the computer, into a digital USB sound card. Therefore, the "mic" in Zoom is configured to be the output of the mixer. Long story short, as long as the KH "mic" is unmuted in Zoom, whatever we hear at the KH is heard over Zoom too. No audio "sharing" necessary. I hope that makes sense?

> QN 2: The quality of the picture depends on the quality of the computer graphics generator machine (don't know what the name is)? I find mine a bit blurry... possibly because I only have the usual onboard graphics thingee that comes with my PC. Will that limit how useful this feature is...?

Make sure your OBS output resolution is set to 720p, and your external screens too. Also, disable auto low lighting adjustment in Zoom. These adjustments made everything crystal clear over Zoom for me.
 
> QN: 3 OBS still needs to be running, but can be put in minimize mode so that it is out of the way of the screen real-estate... is that right?

Correct, I set OBS to auto-start the virtual camera on launch, and to launch minimized by default.

**Reply by Chipperchap** [reply]
> ...Then, all audio from the mixer (mics, computer sound, etc) is sent back to the computer, into a digital USB sound card...

Intriguing.  Very similar setup except: We have a totally analogue mixer and we have the REC OUT returning to the PC. Mics are fine but music is often distorted if the booth personnel forget to ‘share computer audio’. So what you are saying is that if we have the REC OUT (RCA to 3.5mm) going instead first to a USB sound card, and ZOOM using that as a ‘Mic’ then our sound will be solid without need of ‘share’? Does it boost it or clarify it somehow?

Okay, so I made as many changes to OBS as my understanding would allow (with a lot of googling to try figure it out!).
Changes included: 
•	Video Base:1280x720p 
•	Video Output: 1280x720
•	Downscale: Bicubic
•	Transform: “Reset” then ‘Fit to screen”
•	Capture Method: EXGI (to prevent yellow border).
•	Launch: Auto start with Virtual Cam 
•	Launch: Auto minimize mode (we only have one fixed stage facing camera so no need for additional scenes so OBS can be ‘out of the way’)
•	ZOOM Group HD 720p setting enabled

The KH has one of the branch Supplier Arrangement PC’s (for this region) similar to this: 

https://www.pbtech.co.nz/product/EXWKSHP8087425/HP-Elitedesk-800-G2-MINI-A-Grade-OFF-LEASE-Intel-C?qr=popular_related_products 

…but only 4gb ram. I think it needs at least 8gb ram.  

Also our KH interweb is 30mb down / 10mb up (but usually much lower bandwidth than that in reality), so we will just have to see how it goes with the ZOOM group HD test/trial/experiment first before we try anything else. I am in isolation for three more days so I will get to see what the output for us at home will be tomorrow. We won't get to try the OBS virtual cam method until a couple of weeks as I caring for aged parents duties and need to test everything and learn it thoroughly before I train the AV desk personnel. 

Sorry for the long post – I have a sincere interest in these things but often get blank stares from others so thank you for your kind patience @riemsesy @sircharlo @joaochris

**Reply by riemsesy** [reply]
@Chipperchap, hello brother, 8gb is necessary. our neighbour congregation with the dell Optiplex 3070 had to upgrade with an extra 4GB SODIMM else the PC would crash. And they don't use OBS, just zoom and VLC.
All on 720p. instructions even say if bandwidth is a problem, 480p will suffice. it's just a little bit blurry.

**Reply by riemsesy** [reply]
@joaochris thnx. I had to use screen share in stead of window share and / or gameshare full screen app. 
we had issues with screen share a month ago, so that's why I didn't think of it to use it again. But it works perfectly now.

**Reply by joaochris** [reply]
> @joaochris thnx. I had to use screen share in stead of window share and / or gameshare full screen app. we had issues with screen share a month ago, so that's why I didn't think of it to use it again. But it works perfectly now.

@riemsesy Glad it went right! 😅

**Reply by riemsesy** [reply]
https://1drv.ms/v/s!Ar84CcBfxBgNr6BDmkOsq8kT9tG6uA

It works fine in our KH (as test)

**Reply by joaochris** [reply]
> https://1drv.ms/v/s!Ar84CcBfxBgNr6BDmkOsq8kT9tG6uA
> 
> Funciona bem em nosso KH (como teste)

It turned out great! Congratulations! 👏

**Reply by sircharlo** [reply]
> Intriguing.  Very similar setup except: We have a totally analogue mixer and we have the REC OUT returning to the PC. Mics are fine but music is often distorted if the booth personnel forget to ‘share computer audio’. So what you are saying is that if we have the REC OUT (RCA to 3.5mm) going instead first to a USB sound card, and ZOOM using that as a ‘Mic’ then our sound will be solid without need of ‘share’? Does it boost it or clarify it somehow?

Great question honestly; distortion *usually* indicates that the input signal was too loud to start with. I'd make sure that your computer isn't outputting analogue audio at 100%, closer to maybe 80%, and adjusting the computer sound's volume from the mixer. Granted, a multitude of reasons can be behind distortion when going from digital to analog and back to digital again. 

I don't think our cheapo USB sound card does any special processing per se. It's simply set as our computer output (goes to analog mixer) and input (receives mixed-down sound from analog mixer).

> …but only 4gb ram. I think it needs at least 8gb ram.  

At least. 

> Also our KH interweb is 30mb down / 10mb up (but usually much lower bandwidth than that in reality), so we will just have to see how it goes with the ZOOM group HD test/trial/experiment first before we try anything else. 

That's basically what we have too. 25/5, actually, which is worse 😂. Works fine for us, but your mileage may vary. 

> Sorry for the long post – I have a sincere interest in these things but often get blank stares from others so thank you for your kind patience @riemsesy @sircharlo @joaochris 

Always a pleasure!

**Reply by sircharlo** [reply]
> https://1drv.ms/v/s!Ar84CcBfxBgNr6BDmkOsq8kT9tG6uA
> 
> It works fine in our KH (as test)

Very nice! My next commit will implement some error catching (in case OBS is acting up or not launched), and also change the scene in OBS immediately when you select one in JWMMF, **if** the current active scene in OBS is not the one configured as the media scene in JWMMF. This should make camera scene switching even smoother and easier to manage for those on sound booth duty.

**Reply by sircharlo** [reply]
> My next commit will implement some error catching (in case OBS is acting up or not launched), and also change the scene in OBS immediately when you select one in JWMMF, **if** the current active scene in OBS is not the one configured as the media scene in JWMMF. This should make camera scene switching even smoother and easier to manage for those on sound booth duty.

Committed. Feel free to test and let me know thoughts/opinions/feedback.

**Reply by joaochris** [reply]
> > My next commit will implement some error catching (in case OBS is acting up or not launched), and also change the scene in OBS immediately when you select one in JWMMF, **if** the current active scene in OBS is not the one configured as the media scene in JWMMF. This should make camera scene switching even smoother and easier to manage for those on sound booth duty.
> 
> Committed. Feel free to test and let me know thoughts/opinions/feedback.

It's great!!! Thank you for your attention and dedication to this project! 😅😎
Just one thing. Let's assume something. I open JWMMF, I'm in media view, and then I open OBS. Does Websockets have the ability to show and update scenes live, without having to go back?

**Reply by sircharlo** [reply]
> Just one thing. Let's assume something. I open JWMMF, I'm in media view, and then I open OBS. Does Websockets have the ability to show and update scenes live, without having to go back?

I don’t think I’ll implement that; unless once in media presentation mode I “listen” periodically to see if OBS comes online to then connect to it. I’ll think about it.

**Reply by riemsesy** [reply]
> I don’t think I’ll implement that; unless once in media presentation mode I “listen” periodically to see if OBS comes online to then connect to it. I’ll think about it.

I tested this by changing websocket port number, then started JWMMF, error came up, I changed the port to the correct number in OBS, with JWMMF still started and then I was able to change scenes. So if OBS comes up later, you get an error but it will work after OBS started.

**Reply by riemsesy** [reply]
> My next commit will implement some error catching (in case OBS is acting up or not launched), and also change the scene in OBS immediately when you select one in JWMMF, ****if** the current active scene in OBS is not the one configured as the media scene in JWMMF**. This should make camera scene switching even smoother and easier to manage for those on sound booth duty.

I understand that you can't change scenes while playing media, as then is the media scene active?
In my testing I could change OBS scene while playing media.

**Reply by joaochris** [reply]
> > Just one thing. Let's assume something. I open JWMMF, I'm in media view, and then I open OBS. Does Websockets have the ability to show and update scenes live, without having to go back?
> 
> I don’t think I’ll implement that; unless once in media presentation mode I “listen” periodically to see if OBS comes online to then connect to it. I’ll think about it.

All right :). In my opinion it's great

**Reply by sircharlo** [reply]
> I understand that you can't change scenes while playing media, as then is the media scene active?
> 
> In my testing I could change OBS scene while playing media.

But it wasn't immediate, right? The logic I implemented when changing scenes is: 

Is media being presented? (Is the scene configured as media scene currently active in OBS?)
If yes: set the camera scene to the selected scene but don't change the scene yet. When media stops playing, the newly selected camera scene will be activated.
If no: set the camera scene to the selected scene and activate that scene immediately.

If you're seeing different behavior then I missed something 😂

**Reply by joaochris** [reply]
> @riemsesy Is not difficult! The steps I did were:
> 
> 1. First I downloaded the **GIT**: https://git-scm.com/
> 2. Then the **NodeJS LTS version**: https://nodejs.org/en/download/
> 3. Installed both
> 4. Then I opened git bash and put `git clone https://github.com/sircharlo/jw-meeting-media-fetcher`
> 5. Then I navigated to the folder at C:\Users[YOUR USER]\jw-meeting-media-fetcher with the right mouse button and then _Git Bash Here_
> 6. Then `npm i`
>    Wait for it to install all modules
> 7. Then `npm start`
>    After this last command the program starts
> 
> If you want to reopen the program, just follow point 5 and 7. To download project updates with point 5 done, it's `git pull`
> 

@mtdvlpr How can I now run the development version? It helps me 😄

**Reply by mtdvlpr** [reply]
@joaochris, you should have enough information with our [CONTRIBUTING.md](https://github.com/sircharlo/meeting-media-manager/blob/master/CONTRIBUTING.md). If not let me know and I'll update the CONTRIBUTING.md

**Reply by joaochris** [reply]
> @joaochris, you should have enough information with our [CONTRIBUTING.md](https://github.com/sircharlo/meeting-media-manager/blob/master/CONTRIBUTING.md). If not let me know and I'll update the CONTRIBUTING.md

Thanks, I got it :) Now which command do I use to download the latest code updates?

**Reply by mtdvlpr** [reply]
> > @joaochris, you should have enough information with our [CONTRIBUTING.md](https://github.com/sircharlo/meeting-media-manager/blob/master/CONTRIBUTING.md). If not let me know and I'll update the CONTRIBUTING.md
> 
> Thanks, I got it :) Now which command do I use to download the latest code updates?

On your forked repo there's a sync fork button. And then locally, you run `git pull`

**Reply by mtdvlpr** [reply]
And if you're on a different branch than master, you run `git merge origin/master` (or `origin/docs` if you're editing the docs branch) to get the latest changes to that branch.

**Reply by mtdvlpr** [reply]
Next time, make a new discussion instead of repurposing an old one :). Makes it easier to find your comments and makes sure other people who commented on this discussion don't get notified.

**Comment by riemsesy** [comment]
I think the OBS setup should be with the application setup instead of with the Meeting Setup?
Also, the background image is more of a media setup. 
The enable button to present media on an external monitor is more a media setup. 

The option to enable to play Kingdom songs on shuffle would fit better directly underneath the meeting time setup because they work together.

just some thoughts.
It's 07:40 AM, will go walk with the dog and then to the KH :-)

Thank you so far, Charles.

**Reply by sircharlo** [reply]
Excellent points. I kind of just dumped everything there in a hurry 😂 but I agree with you.

**Reply by sircharlo** [reply]
Done; committed.

**Comment by riemsesy** [comment]
Wouldn’t a reload button in the config part be enough? Scenes in OBS don’t change that much. Mostly I keep the scenes names the same and just change the sources.

Groeten, Richard
________________________________
Van: joaochris ***@***.***>
Verzonden: Sunday, May 1, 2022 9:29:12 AM
Aan: sircharlo/jw-meeting-media-fetcher ***@***.***>
CC: riemsesy ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [sircharlo/jw-meeting-media-fetcher] Virtual camera output (Discussion #270)


Just one thing. Let's assume something. I open JWMMF, I'm in media view, and then I open OBS. Does Websockets have the ability to show and update scenes live, without having to go back?

I don’t think I’ll implement that; unless once in media presentation mode I “listen” periodically to see if OBS comes online to then connect to it. I’ll think about it.

All right :). In my opinion, that's great!

**Reply by joaochris** [reply]
@riemsesy Exactly! Just leave all the scenes created previously. And as you said we can change the sources. Studio mode makes it easy to make changes live without affecting anything, so it's smooth sailing.

**Reply by sircharlo** [reply]
Sorry I missed a step, what is question this in relation to?

**Reply by riemsesy** [reply]
The question was if JWMMF will poll automatically if obs is live and what the scene names are. But it isn't really necessary. Scene names don't change that often. If this feature is necessary a manual reload button would be enough.

**Reply by joaochris** [reply]
> Desculpe, eu perdi uma etapa, o que é essa pergunta em relação?

The idea he came up with was to put a refresh button, which would update the OBS scenes. It relates to what I said about Websockets' ability to update scenes in the media view.

**Reply by sircharlo** [reply]
Yeah, if this becomes a need I could add it eventually, but I suggest setting up scenes beforehand and then controlling them with JWMMF. I don't think the AV operator will be creating scenes on the fly at every meeting. Seems like overkill to me. 😆

**Reply by Uilsonjw** [reply]
I'm writing here because my idea looks related with this post. Would be good if JWMMF refresh the combo when the scenes are loaded or if the scene is changed in OBS. 

Why it's desirable? 
1) In OBS we can set shurtcuts to load a specifc scene, so if the operator uses the shortcut, the combo would be refreshed.

2) We could assign another brother to change the cameras scene by using a remote software like OBS Blade for Android or iOS, so if the scene is changed by the remote software, the combo woud be updated.

So, would be nice if once the scene is changed in the OBS, the JWMMF refreshs the combo with it.

**Reply by sircharlo** [reply]
I added scene change "listening" [here](https://github.com/sircharlo/jw-meeting-media-fetcher/commit/3cc4754d9dd3db4ff67d888c753e18f344b0b24b). Basically, if the active scene changes in OBS Studio, that change will be reflected in JWMMF.

**Comment by Chipperchap** [comment]
@riemsesy - Please, what kind of camera is your KH using? Please can you provide a link so I can view the type it is? 🙏

@sircharlo - Sorry, I know you are busy - you said you have a "cheapo USB sound card" - please do you have a link I can view what kind it is? 🙏

**Reply by sircharlo** [reply]
I'll see if I can grab that info for you when I go to the hall Wednesday.

**Reply by riemsesy** [reply]
Jehttps://minriglory.com/products-item/uv570-series-hd-video-conference-camera/

We have 20x zoom. Connected with coax SDI. 
To connect coax to the pc you need a pcie card with coax connector. 

https://www.blackmagicdesign.com/products/decklink/techspecs/W-DLK-06

Or

https://www.activesilicon.com/news-media/news/solutions-for-sdi-usb-video-capture/

Or 

https://www.usbfever.com/bnc-to-usb-video-converter-cable

Of course of the distance is not too long you can also use hdmi. 

Manage the camera via utp network. 

NDI protocol also works via Ethernet.

**Reply by Chipperchap** [reply]
> Manage the camera via utp network.

Ok - So I bought a SDI 20x ZOOM PTZ camera of the type suggested by the branch in this region. (Arrived). I want to get everything working before I attempt to install in the K-Hall. (Pending approval which is 1/2 way there... just waiting on our sister congregation).
I also bought a SDI to USB 3.1 Type-C adapter/converter for the image side of things, and a PoE switch to power the camera over ethernet and to 'control' it. (both appliances pending arrival / in transit).   What I really need now is a tutorial on how all this works with OBS / M3.
I know how to create scenes / sources in OBS, and how these are shown as available to select in M3.  But what I don't understand at this point is how OBS tells the PoE Switch what to then tell the camera what to do.  Is there some kind of IP address thingee that I need to configure? How does one do this? (The documentation that came with the camera is v-e-r-y basic.) 
The PTZ camera address is 192.168.1.123.
Soooo, as quoted above, how to do I "manage the camera via utp network"...?

Incidentally I have been asked by LDC in this country to assist other congregations in the area on demand, so want to get it right at home first before I go elsewhere. 

Any help would be deeply appreciated. 

Kind regards
Simon

**Reply by riemsesy** [reply]
Well. UTP is just the type cable. Normally we just call it network or Ethernet. 
The cam has an IP number. Your pc has an ip number. They must be in the same network. 
As example:
Cam: 192.168.1.10
Your pc: 192.168.1.100
If they're not on the same network you won't be able to connect to the cam. 
When they are in the same network then just type in the ip number of the camera in the browser. Check the manual if you need to add a port number like http://192.168.1.10:5000 (example)

The switch doesn't do anything with the network addresses. It just passes on data packages. And in case of Poe it powers a device. 

When you're not familiar with this, then find someone locally to help you with this. 
It's not easy to learn someone remote via text about networking. 
It can mean to know how to change the IP number of the PC, change dhcp settings in the router(sometimes modem), to scan the network for devices. 

As start, you don't need the cam connected with SD to make connection with it. Set up the network connection first and later try to capture video image via SD cable on your pc.

**Reply by mtdvlpr** [reply]
To let OBS connect to your camera you can use the [ptz-controls plugin](https://obsproject.com/forum/resources/ptz-controls.1284/). Once configured, you can add a ptz-controls action as a source to every scene (scene Speaker with ptz-action "go to preset 1" for example).

**Reply by Chipperchap** [reply]
Thank you @mtdvlpr - I worked it out. One day I will show my results.

**Reply by riemsesy** [reply]
Great to hear it works 👍🏻

**Comment by sircharlo** [comment, marked as answer]
Take a look at the updated [documentation](https://sircharlo.github.io/meeting-media-manager/en/#present-media), I'd be curious to get feedback from all of you now that the latest version is out.

**Reply by joaochris** [reply]
> Take a look at the updated [documentation](https://sircharlo.github.io/jw-meeting-media-fetcher/#/present-media), I'd be curious to get feedback from all of you now that the latest version is out.

😎 A question. Why do you need to install the virtual cam plugin if it already comes with OBS? 🤔

**Reply by sircharlo** [reply]
I went with the portable version of OBS Studio, I think that's why by default it wasn't installed/enabled for me; some users indicate that the plugin works a bit better for them than the integrated version. 🤷

**Reply by joaochris** [reply]
> I went with the portable version of OBS Studio, I think that's why by default it wasn't installed/enabled for me; some users indicate that the plugin works a bit better for them than the integrated version. 🤷

Oh yes, I understand. Today at the meeting with the _CO_, I will use the integration. But for me it's great! It's very simple!  😀😎
